### PR TITLE
SAK-44301 refactor and streamline Forums permission checking

### PR DESCRIPTION
**Pull request by ottenhoff**
Before patch with 200+ groups and 500+ topics: 14.65 sec as instructor and 6.02 sec as student
After patch with 200+ groups and 500+ topics: 8.04 sec as instructor and 1.86 sec as student

### CODE REVIEW

**Comment by adrianfish** [comment]
Can you squash it? I can't handle these multi commit PRs, drive me nutty.

**Comment by ottenhoff** [comment]
i dont understand why. just squash on merge

**Comment by adrianfish** [comment]
I was going to review it! I mean when reviewing.

**Comment by ottenhoff** [comment]
Files changed tab shows all changes in this PR .....

File: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionTopicBean.java [hunk 1]
```diff
@@ -1133,14 +1134,17 @@ public List recursivelyGetUnreadMessagesInThreads(List curList, int depth){
...
+			List<Attachment> attachList = forumManager.getTopicAttachments(topic.getId());  
+			if(attachList != null)
 			{
-				DecoratedAttachment decoAttach = new DecoratedAttachment((Attachment)attachList.get(i));
-				decoAttachList.add(decoAttach);
+				for(int i=0; i<attachList.size(); i++)
+				{
+					DecoratedAttachment decoAttach = new DecoratedAttachment((Attachment)attachList.get(i));
+					decoAttachList.add(decoAttach);
+				}
```
Comment (bjones86) [inline msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/ui/DiscussionTopicBean.java:1147]: I think you should be able to do this in a one line stream operation.

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/DiscussionForumManagerImpl.java [hunk 2]
```diff
@@ -2191,87 +2203,118 @@ public DBMembershipItem getDBMember(Set originalSet, String name,
...
-    		else{    	
-    			/** get level from config file */
-    			levelName = ServerConfigurationService.getString(MC_DEFAULT
-              + name);
-    			    			
-    			
-    		}
+    	String levelName = null;
+ 
+    	if (type.equals(DBMembershipItem.TYPE_ROLE)) {
```
Comment (bjones86) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/DiscussionForumManagerImpl.java:2270]: Is type guaranteed not null? You could flip the operands here for null safety.

File: msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java [hunk 3]
```diff
@@ -411,17 +444,17 @@ public boolean isMovePostings(DiscussionTopic topic, DiscussionForum forum)
...
       {
         DBMembershipItem item = (DBMembershipItem) iter.next();
-        if ((item.getPermissionLevel().getMovePosting().booleanValue()
-            || item.getPermissionLevel().getReviseAny().booleanValue()
-            || item.getPermissionLevel().getReviseOwn().booleanValue())
-            && forum.getDraft().equals(Boolean.FALSE)
-            && forum.getLocked().equals(Boolean.FALSE)
-            && topic.getDraft().equals(Boolean.FALSE)
-            && topic.getLocked().equals(Boolean.FALSE))
+        if (item.getPermissionLevel().getMovePosting().booleanValue() || item.getPermissionLevel().getReviseAny().booleanValue() || item.getPermissionLevel().getReviseOwn().booleanValue())
```
Comment (bjones86) [inline msgcntr/messageforums-component-impl/src/java/org/sakaiproject/component/app/messageforums/ui/UIPermissionsManagerImpl.java:457]: Probably not necessary to unbox these now.

**Review by bjones86 — APPROVED** [review]
Took a quick look. LGTM, just some minor questions.

**Comment by bjones86** [comment]
Oh, we got some conflicts on here now.

**Pull request closed without merging**